lib/graphql/batch/execution_strategy.rb

```diff
@@ -1,66 +0,0 @@
-module GraphQL::Batch
-  class ExecutionStrategy < GraphQL::Query::SerialExecution
-    attr_accessor :disable_batching
-
-    def execute(_, _, query)
-      as_promise(super).sync
-    rescue GraphQL::InvalidNullError => err
-      err.parent_error? || query.context.errors.push(err)
-      nil
-    ensure
-      GraphQL::Batch::Executor.current.clear
```

Comment:
Looks like we might need to caution users about how state can now leak between requests without an ensure block like this.

For example, if some loader doesn't get loaded, then that load could happen on a following query.  Also, since loads a batched, it is possible for a specific key to cause a loader to raise an exception, causing a following request to fail from a load in a previous request.